app/Controllers/CtrlSolution.php

```diff
@@ -20,114 +39,176 @@ public function viewSolutions()
 
     public function viewSolutionCreate()
     {
-        return view('admin/typeSolutions/SolutionCreate');
+        if (session()->has('clientData')) {
+            $this->fileConfig->setDataInClientConfig(session()->get('clientData'));
+        }
+
+        return view('admin/typeSolutions/SolutionCreate', ['filepondConfig' => $this->fileConfig->getClientConfig()]);
     }
 
-    public function viewSolutionEdit($id)
+    public function viewSolutionEdit($msId)
     {
-        $measurementSolutionModel = new MeasurementSolutionModel();
-        $measurementSolution      = $measurementSolutionModel->find($id);
+        $msData = (new MeasurementSolutionModel())->getMeasurementSolutionDataWithFiles($msId);
+
+        if (! session()->has('clientData')) {
+            $dataFiles['msIcon']  = [$msData['msIcon']];
+            $dataFiles['msImage'] = [$msData['msImage']];
+        } else {
+            $dataFiles = session()->get('clientData');
+        }
```

Comment:
Revert the if always use the true path first
```suggestion
        if (session()->has('clientData')) {
            $dataFiles = session()->get('clientData');
        } else {
            $dataFiles['msIcon']  = [$msData['msIcon']];
            $dataFiles['msImage'] = [$msData['msImage']];
        }
```